Article: Interview with robjtede (#84)
RustMagazine/rustmagazine: content/issue-3/robjtede.md

> We are thrilled to present our third interview with **Rob** ([@robjtede]), the core maintainer of [Actix Web](https://github.com/actix/actix-web) and [deps.rs](https://deps.rs/). In this interview, we talked about how he became involved with Actix Web, what motivated him to take over the project after [@fafhrd91] quit, and his experience of maintaining popular Rust projects.

Author: Good point, thanks.

Thanks to **Rob** for sharing his story with us. We hope you enjoy this interview!

![](/static/issue-3/robjtede.jpeg)

> > Rob and a Saitama cosplay from One Punch Man, and his avatar also is Saitama.

Reviewer: just a suggestion:

```suggestion
> Rob uses One Punch Man's Saitama as his avatar online. Here he is meeting his hero.
```

I found out about Rust at some point during university and started [practicing it during 2017's Advent of Code](https://github.com/robjtede/adventofcode2017) event. It was more or less just a curiosity at that point, as my main development was focussed on JavaScript and other frontend skills plus some Node.JS which I used in my undergraduate dissertation project.

It wasn't until I started working (a frontend role) that I started using Rust more. I had suggested to the (small) team that part of the monolithic application we were creating could be split out. It was written in Java, and I had no intention of writing Java so Rust seemed like a good idea to inject into our stack. Most of the team had at least heard of Rust and understood its stated benefits so they gave me free license to develop this service alongside my normal frontend work.

In the 4 years since then, I've transitioned to a purely backend role; as the team grew I needed to focus more on one side of the stack in order to keep my skills sharp. This lead to a desire to go deeper into Rust and its ecosystem.
"""
```

# Actix Web

> **How did you become involved with actix-web, and what motivated you to take over the project after [@fafhrd91] quit?**

Reviewer: ```suggestion
**How did you become involved with Actix Web, and what motivated you to take over the project after [@fafhrd91] quit?**
```

During this time, I'd been driving the development work along with some other contributors, and **Yuki** had been approving my PRs diligently. At some point he must have realized I was serious about maintaining Actix Web and he granted me owner permissions to the repo and publish permissions to the crates.io registry.
"""
```

> **What is the current status of actix-web, and how does it compare to two years ago? Are there any major changes or updates in the works?**

Reviewer: ```suggestion
**What is the current status of Actix Web, and how does it compare to two years ago? Are there any major changes or updates in the works?**
```

As for future changes, there is certainly a v5 on the cards. The most exciting reasons for a major version bump are the changes we'll be able to make to key traits like `FromRequest` now that Generic Associated Types (GATs) have been stabilized and return-position impl-trait and/or `async fn` in traits seems to be (perpetually) around the corner. These features will make writing custom extractors and middleware much more pleasant for our users (and us).

Some of the less exciting breaking changes are being trialled in the [`actix-web-lab`](https://crates.io/crates/actix-web-lab) crate, a place where I have been busy experimenting, unburdened by the burden of post-v1 semantic versioning. (This crate has helped me massively on the API stability goal of v4.x, too. Lots of cool, non-breaking features are written for this crate and then graduate to Actix Web after seeing some real world usage and having their sharp edges ironed out.)
"""
```

> **Is actix-web still a web framework powered by the actor model? If not, why?**

Reviewer: ```suggestion
**Is Actix Web still a web framework powered by the actor model? If not, why?**
```

> We are thrilled to present our third interview with **Rob** ([@robjtede]), the core maintainer of [actix-web](https://github.com/actix/actix-web) and [deps.rs](https://deps.rs/). In this interview, we talked about how he became involved with actix-web, what motivated him to take over the project after [@fafhrd91] quit, and his experience of maintaining popular Rust projects.

Reviewer: ```suggestion
We are thrilled to present our third interview with **Rob** ([@robjtede]), the core maintainer of [Actix Web](https://github.com/actix/actix-web) and [deps.rs](https://deps.rs/). In this interview, we talked about how he became involved with actix-web, what motivated him to take over the project after [@fafhrd91] quit, and his experience of maintaining popular Rust projects.
```

I found out about Rust at some point during university and started [practicing it during 2017's Advent of Code](https://github.com/robjtede/adventofcode2017) event. It was more or less just a curiosity at that point, as my main development was focussed on JavaScript and other frontend skills plus some Node.JS which I used in my undergraduate dissertation project.

It wasn't until I started working (a frontend role) that I started using Rust more. I had suggested to the (small) team that part of the monolithic application we were creating could be split out. It was written in Java, and I had no intention of writing Java so Rust seemed like a good idea to inject into our stack. Most of the team had at least heard of Rust and understood its stated benefits so they gave me free license to develop this service alongside my normal frontend work.

In the 4 years since then, I've transitioned to a purely backend role; as the team grew I needed to focus more on one side of the stack in order to keep my skills sharp. This lead to a desire to go deeper into Rust and its ecosystem.
"""
```

> # actix-web

Reviewer: Can "Actix Web" always be formatted like this, please, ie. capitalized and no hyphen:

```suggestion
# Actix Web
```
